src/aiida_quantumespresso/calculations/functions/seekpath_structure_analysis.py

```diff
@@ -28,4 +30,47 @@ def seekpath_structure_analysis(structure, **kwargs):
     # All keyword arugments should be `Data` node instances of base type and so should have the `.value` attribute
     unwrapped_kwargs = {key: node.value for key, node in kwargs.items() if isinstance(node, Data)}
 
-    return get_explicit_kpoints_path(structure, **unwrapped_kwargs)
+    result = get_explicit_kpoints_path(structure, **unwrapped_kwargs)
+
+    # If the input structure was a HubbardStructureData, update the primitive structure with Hubbard parameters
+    if isinstance(structure, HubbardStructureData):
+        update_structure_with_hubbard(structure, result)
+
+    return result
+
+
+def update_structure_with_hubbard(structure, result):
+    """Update the structure based on Hubbard parameters if the input structure is a HubbardStructureData."""
+    hubbard_parameters = structure.hubbard.parameters
+    if not hubbard_parameters:
+        return
+
+    hubbard_structure = HubbardStructureData.from_structure(result['primitive_structure'])
+
+    for parameter in hubbard_parameters:
+        atom_index = parameter.atom_index
+        atom_name = structure.sites[atom_index].kind_name
+
+        if parameter.hubbard_type != 'V':
+            hubbard_structure.initialize_onsites_hubbard(
+                atom_name=atom_name,
+                atom_manifold=parameter.atom_manifold,
+                value=parameter.value,
+                hubbard_type=parameter.hubbard_type,
+                use_kinds=True,
+            )
+        else:
```

Comment:
Here we should add the `NotImplementedError` with a clear message to the user or implement a generalized function(s) for V

Reply:
I know. ^^ Just for future reference: I typically leave a PR in "Draft" status until it's ready for proper review, and request a review at that point. 🙃

Reply:
ah yeah i now , i was putting a comment here just as a reminder , sorry :)